### ISSUE TITLE
Detect control files and automatically check for correspondence

### ISSUE THREAD

**Issue by Breizhux** [gtkhash/gtkhash#165]
Hello,

**Would it be possible to implement a function that automatically detects the checksum file associated with a given file, opens it, extracts the string corresponding to the checksum and automatically compares it to the checksum file?**

Let me explain:

Suppose I open a "linux.iso" file to calculate its checksum.
Let's imagine that next to it is the file "linux.iso.sha256" or "linux.iso.sha256sum", which is not uncommon, since the two files are often downloaded from the same web page, with the two links next to each other.
It would be very useful to automatically detect whether such a file exists next to the open file. Open it, extract the result (often files contain [result sha256]\t[filename, here it would be "linux.iso"] or the reverse order [linux.iso]\t[result sha256].

In short, adding a condition of this style to the opening of a :
```
input_file = "linux.iso"
control_file = ""
if input_file+".sha256" exists :
    control_file = input_file+".sha256"
if input_file+".sha256sum" exists :
    control_file = input_file+".sha256sum"
if control_file != "" :
    content = read(control_file).split("\t")
    if content[0] == input_file :
        digest = content[1]
    else :
        digest = content[0]
    #when calcul input_file digest, verify if corresponding to readed digest, without others manipulations users.
```

**Thank you for everything you can do!**

**Comment by Breizhux** [comment]
**Why do in 3 clicks what could be done in 1?**
_Isn't it very complicated to call the function that hides behind the "Open..." button in the "File" menu when the same file name exists as the one being opened, followed by an extension such as ".sha256" or ".md5sum"?_

I'm not asking you to replace the existing function. I'm just asking you to add this automatic feature that I'm sure a lot of people are waiting for!

It would also be very useful if this automatic check could be performed in the GtkHash tab in the file properties! This would avoid all the following steps:
 - changing windows (bringing Caja back to the foreground, for example)
 - open the file next to it that ends with `.sha256`
 - copy the desired character string
 - close the window (with possible loss of the clipboard depending on the default editor)
 - return to the file properties window
 - finally, paste the checksum...

**All this could be automated with a single click!**

**Comment by Breizhux** [comment]
Oh right, I hadn't quite understood.
**Great, thank you so much!** This will be my Christmas present, as well as a lot of other people's ;)
